tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -1913,6 +1913,579 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultiStageElecHeatCoil_Only)
+    EXPECT_GT(thisSys->m_HeatingSpeedRatio, 0.0);
+    // supp coil doesn't need to turn on
+    EXPECT_EQ(0, thisSys->m_SuppHeatingSpeedNum);
+    EXPECT_EQ(0.0, thisSys->m_SuppHeatPartLoadFrac);
+    EXPECT_EQ(0.0, thisSys->m_SuppHeatingCycRatio);
+    EXPECT_EQ(0.0, thisSys->m_SuppHeatingSpeedRatio);
+
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand(ControlZoneNum).RemainingOutputRequired = 5500;
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand(ControlZoneNum).RemainingOutputReqToHeatSP = 5500;
+    // Heating mode
+    thisSys->simulate(*state,
+                      thisSys->Name,
+                      FirstHVACIteration,
+                      AirLoopNum,
+                      CompIndex,
+                      HeatActive,
+                      CoolActive,
+                      ZoneOAUnitNum,
+                      OAUCoilOutTemp,
+                      ZoneEquipment,
+                      sensOut,
+                      latOut);
+    // main heating coil is meeting demand
+    EXPECT_NEAR(sensOut, 5500, 0.001 * 7500);
+    EXPECT_EQ(sensOut, thisSys->m_SensibleLoadMet);
+    EXPECT_EQ(thisSys->m_SensHeatEnergyRate, thisSys->m_SensibleLoadMet);
+    EXPECT_EQ(5500.0, thisSys->m_HeatingCoilSensDemand);
+    // Main heating coil is operating between speed1 and speed2
+    EXPECT_EQ(2, thisSys->m_HeatingSpeedNum);
+    EXPECT_EQ(1.0, thisSys->m_HeatingCycRatio);
+    EXPECT_EQ(thisSys->m_HeatingPartLoadFrac, 1.0);
+    EXPECT_EQ(thisSys->m_HeatingSpeedRatio, 1.0);
+    // supp coil at 1st speed
+    EXPECT_EQ(1, thisSys->m_SuppHeatingSpeedNum);
+    EXPECT_GT(thisSys->m_SuppHeatPartLoadFrac, 0.0);
+    EXPECT_GT(thisSys->m_SuppHeatingCycRatio, 0.0);
+    EXPECT_EQ(0.0, thisSys->m_SuppHeatingSpeedRatio);
+
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand(ControlZoneNum).RemainingOutputRequired = 8000;
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand(ControlZoneNum).RemainingOutputReqToHeatSP = 8000;
+    // Heating mode
+    thisSys->simulate(*state,
+                      thisSys->Name,
+                      FirstHVACIteration,
+                      AirLoopNum,
+                      CompIndex,
+                      HeatActive,
+                      CoolActive,
+                      ZoneOAUnitNum,
+                      OAUCoilOutTemp,
+                      ZoneEquipment,
+                      sensOut,
+                      latOut);
+    // main heating coil is meeting demand
+    EXPECT_NEAR(sensOut, 8000, 0.001 * 10000);
+    EXPECT_EQ(sensOut, thisSys->m_SensibleLoadMet);
+    EXPECT_EQ(thisSys->m_SensHeatEnergyRate, thisSys->m_SensibleLoadMet);
+    EXPECT_EQ(8000.0, thisSys->m_HeatingCoilSensDemand);
+    // Main heating coil is operating between speed1 and speed2
+    EXPECT_EQ(2, thisSys->m_HeatingSpeedNum);
+    EXPECT_EQ(1.0, thisSys->m_HeatingCycRatio);
+    EXPECT_EQ(thisSys->m_HeatingPartLoadFrac, 1.0);
+    EXPECT_EQ(thisSys->m_HeatingSpeedRatio, 1.0);
```

Comment:
The comment above (line 2215) says "// Main heating coil is operating between speed1 and speed2" but these tests show the main heating coil is max'd out at speed = 2 and heating speed ratio = 1. And below the supplemental heater picks up the remaining load. Line 2220 say "// supp coil at 1st speed" but line 2221 shows m_SuppHeatingSpeedNum = 2 ?

Reply:
Sorry, the comments are outdated, pushing the updates